lib/relationships.js

```diff
@@ -98,6 +93,36 @@ function relationships (id, data, model) {
 
         updateRelation(modelTo, idToFind, where)
       })
+
+      if (serverRelation.modelThrough) {
+        var modelThrough = serverRelation.modelThrough
+        var key = keyByModel(modelThrough, modelTo)
+        var data = {}
+        data[fkName] = id
+
+        var payloadIds = _.map(relationship.data, function (item) {
+          return item.id
+        })
+
+        modelThrough.find({where: data}, function (err, instances) {
+          if (err) return
+          var serverIds = _.map(instances, function (instance) {
+            return instance[key]
+          })
+          // to delete
+          var toDelete = _.difference(serverIds, payloadIds)
```

Comment:
I have tested it with a project. `serverIds` are numbers, but Ember sends string ids so the `_.difference` will give wrong result. I solved it with casting to string.
```Javascript
         var serverIds = _.map(instances, function (instance) {
            return instance[key].toString()
          })
```
I think the component should be check for type differences and cast only when it is necessary. Later I am gonna make some test for it, and fix it.